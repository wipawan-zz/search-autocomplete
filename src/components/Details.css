.details-container {
  width: 700px;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: space-between;
}

.details-text-container {
  text-align: left;
  max-width: 500px;
  word-wrap: break-word;
}

.details-repo-title {
  font-size: 36px;
}

.details-repo-title:hover {
  text-decoration: underline;
}

.details-repo-text {
  padding-top: 0.5em;
}

.details-repo-desc {
  padding-top: 0.5em;
  color: #858585;
}

.details-icons {
  width: 200px;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: space-around;
}

.details-icon-group {
  display: inline-block;
}

.details-icon-item {
  display: block;
  padding-bottom: 0.1em;
}