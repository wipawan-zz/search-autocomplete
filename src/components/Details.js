import React from 'react';
import IconsGroup from './IconsGroup';

import './Details.css';


const Details = ({ selectedItem }) => (
  <div className="details-container">
    <div className="details-text-container">
      <div className="details-repo-title" onClick={() => window.open(selectedItem.html_url, "_blank")}>{selectedItem.full_name}</div>
      <div className="details-repo-text">{selectedItem.language}</div>
      <div className="details-repo-desc">{selectedItem.description}</div>
    </div>
    <IconsGroup 
      stars={selectedItem.stargazers_count}
      forks={selectedItem.forks_count}
      watchers={selectedItem.watchers_count}
    />
  </div>
);

export default Details;