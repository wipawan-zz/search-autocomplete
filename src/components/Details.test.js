import React from 'react';
import { shallow } from 'enzyme';

import Details from './Details';

import { stateMock } from '../stateMock';

describe('<Details />', () => {
  test('renders the component', () => {
    const details = shallow(<Details selectedItem={stateMock.selectedItem} />);

    expect(details).toMatchSnapshot();
  });

  describe('onClick', () => {
    test('shows correct repository title', () => {
      const details = shallow(<Details selectedItem={stateMock.selectedItem} />);
  
      expect(details.find('.details-repo-title').getElement().props.children).toMatchSnapshot();
    });

    test('correctly calls click handler', () => {
      const details = shallow(<Details selectedItem={stateMock.selectedItem} />);
      global.open = jest.fn();

      details.find('.details-repo-title').simulate('click');
  
      expect(global.open).toBeCalled();;
    });
  });
});