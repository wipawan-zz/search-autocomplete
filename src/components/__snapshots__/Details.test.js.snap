// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> onClick shows correct repository title 1`] = `"d3/d3"`;

exports[`<Details /> renders the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Details
    selectedItem={
        Object {
            "archive_url": "https://api.github.com/repos/d3/d3/{archive_format}{/ref}",
            "archived": false,
            "assignees_url": "https://api.github.com/repos/d3/d3/assignees{/user}",
            "blobs_url": "https://api.github.com/repos/d3/d3/git/blobs{/sha}",
            "branches_url": "https://api.github.com/repos/d3/d3/branches{/branch}",
            "clone_url": "https://github.com/d3/d3.git",
            "collaborators_url": "https://api.github.com/repos/d3/d3/collaborators{/collaborator}",
            "comments_url": "https://api.github.com/repos/d3/d3/comments{/number}",
            "commits_url": "https://api.github.com/repos/d3/d3/commits{/sha}",
            "compare_url": "https://api.github.com/repos/d3/d3/compare/{base}...{head}",
            "contents_url": "https://api.github.com/repos/d3/d3/contents/{+path}",
            "contributors_url": "https://api.github.com/repos/d3/d3/contributors",
            "created_at": "2010-09-27T17:22:42Z",
            "default_branch": "master",
            "deployments_url": "https://api.github.com/repos/d3/d3/deployments",
            "description": "Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:",
            "downloads_url": "https://api.github.com/repos/d3/d3/downloads",
            "events_url": "https://api.github.com/repos/d3/d3/events",
            "fork": false,
            "forks": 19552,
            "forks_count": 19552,
            "forks_url": "https://api.github.com/repos/d3/d3/forks",
            "full_name": "d3/d3",
            "git_commits_url": "https://api.github.com/repos/d3/d3/git/commits{/sha}",
            "git_refs_url": "https://api.github.com/repos/d3/d3/git/refs{/sha}",
            "git_tags_url": "https://api.github.com/repos/d3/d3/git/tags{/sha}",
            "git_url": "git://github.com/d3/d3.git",
            "has_downloads": true,
            "has_issues": false,
            "has_pages": false,
            "has_projects": false,
            "has_wiki": true,
            "homepage": "https://d3js.org",
            "hooks_url": "https://api.github.com/repos/d3/d3/hooks",
            "html_url": "https://github.com/d3/d3",
            "id": 943149,
            "issue_comment_url": "https://api.github.com/repos/d3/d3/issues/comments{/number}",
            "issue_events_url": "https://api.github.com/repos/d3/d3/issues/events{/number}",
            "issues_url": "https://api.github.com/repos/d3/d3/issues{/number}",
            "keys_url": "https://api.github.com/repos/d3/d3/keys{/key_id}",
            "labels_url": "https://api.github.com/repos/d3/d3/labels{/name}",
            "language": "JavaScript",
            "languages_url": "https://api.github.com/repos/d3/d3/languages",
            "license": Object {
              "key": "bsd-3-clause",
              "name": "BSD 3-Clause \\"New\\" or \\"Revised\\" License",
              "node_id": "MDc6TGljZW5zZTU=",
              "spdx_id": "BSD-3-Clause",
              "url": "https://api.github.com/licenses/bsd-3-clause",
            },
            "merges_url": "https://api.github.com/repos/d3/d3/merges",
            "milestones_url": "https://api.github.com/repos/d3/d3/milestones{/number}",
            "mirror_url": null,
            "name": "d3",
            "node_id": "MDEwOlJlcG9zaXRvcnk5NDMxNDk=",
            "notifications_url": "https://api.github.com/repos/d3/d3/notifications{?since,all,participating}",
            "open_issues": 6,
            "open_issues_count": 6,
            "owner": Object {
              "avatar_url": "https://avatars1.githubusercontent.com/u/1562726?v=4",
              "events_url": "https://api.github.com/users/d3/events{/privacy}",
              "followers_url": "https://api.github.com/users/d3/followers",
              "following_url": "https://api.github.com/users/d3/following{/other_user}",
              "gists_url": "https://api.github.com/users/d3/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/d3",
              "id": 1562726,
              "login": "d3",
              "node_id": "MDEyOk9yZ2FuaXphdGlvbjE1NjI3MjY=",
              "organizations_url": "https://api.github.com/users/d3/orgs",
              "received_events_url": "https://api.github.com/users/d3/received_events",
              "repos_url": "https://api.github.com/users/d3/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/d3/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/d3/subscriptions",
              "type": "Organization",
              "url": "https://api.github.com/users/d3",
            },
            "permissions": Object {
              "admin": false,
              "pull": true,
              "push": false,
            },
            "private": false,
            "pulls_url": "https://api.github.com/repos/d3/d3/pulls{/number}",
            "pushed_at": "2018-06-21T16:53:20Z",
            "releases_url": "https://api.github.com/repos/d3/d3/releases{/id}",
            "score": 86.35898,
            "size": 35963,
            "ssh_url": "git@github.com:d3/d3.git",
            "stargazers_count": 77640,
            "stargazers_url": "https://api.github.com/repos/d3/d3/stargazers",
            "statuses_url": "https://api.github.com/repos/d3/d3/statuses/{sha}",
            "subscribers_url": "https://api.github.com/repos/d3/d3/subscribers",
            "subscription_url": "https://api.github.com/repos/d3/d3/subscription",
            "svn_url": "https://github.com/d3/d3",
            "tags_url": "https://api.github.com/repos/d3/d3/tags",
            "teams_url": "https://api.github.com/repos/d3/d3/teams",
            "trees_url": "https://api.github.com/repos/d3/d3/git/trees{/sha}",
            "updated_at": "2018-07-31T03:10:43Z",
            "url": "https://api.github.com/repos/d3/d3",
            "watchers": 77640,
            "watchers_count": 77640,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="details-text-container"
>
          <div
                    className="details-repo-title"
                    onClick={[Function]}
          >
                    d3/d3
          </div>
          <div
                    className="details-repo-text"
          >
                    JavaScript
          </div>
          <div
                    className="details-repo-desc"
          >
                    Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:
          </div>
</div>,
        <IconsGroup
          forks={19552}
          stars={77640}
          watchers={77640}
/>,
      ],
      "className": "details-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="details-repo-title"
              onClick={[Function]}
>
              d3/d3
</div>,
            <div
              className="details-repo-text"
>
              JavaScript
</div>,
            <div
              className="details-repo-desc"
>
              Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:
</div>,
          ],
          "className": "details-text-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "d3/d3",
              "className": "details-repo-title",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "d3/d3",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "JavaScript",
              "className": "details-repo-text",
            },
            "ref": null,
            "rendered": "JavaScript",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:",
              "className": "details-repo-desc",
            },
            "ref": null,
            "rendered": "Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "forks": 19552,
          "stars": 77640,
          "watchers": 77640,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="details-text-container"
>
            <div
                        className="details-repo-title"
                        onClick={[Function]}
            >
                        d3/d3
            </div>
            <div
                        className="details-repo-text"
            >
                        JavaScript
            </div>
            <div
                        className="details-repo-desc"
            >
                        Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:
            </div>
</div>,
          <IconsGroup
            forks={19552}
            stars={77640}
            watchers={77640}
/>,
        ],
        "className": "details-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="details-repo-title"
                onClick={[Function]}
>
                d3/d3
</div>,
              <div
                className="details-repo-text"
>
                JavaScript
</div>,
              <div
                className="details-repo-desc"
>
                Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:
</div>,
            ],
            "className": "details-text-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "d3/d3",
                "className": "details-repo-title",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "d3/d3",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "JavaScript",
                "className": "details-repo-text",
              },
              "ref": null,
              "rendered": "JavaScript",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:",
                "className": "details-repo-desc",
              },
              "ref": null,
              "rendered": "Bring data to life with SVG, Canvas and HTML. :bar_chart::chart_with_upwards_trend::tada:",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "forks": 19552,
            "stars": 77640,
            "watchers": 77640,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
